### PR TITLE
Refactor formulating index.html file logic so we can use Ngrok to tes…

### PR DESCRIPTION
**Pull request by HanKyungSung**
We couldn't test og tags and other meta tags from the development environment. 

This PR includes the new route `test-subdomain/:subdomain` that we can use this route to test og tags and other meta tags.

> Note: Ngrok provides paid service with a subdomain but since we are using the free version, we don't have access to that feature so we decide to use a new route.

### CODE REVIEW

File: invitation-backend/src/app.ts [hunk 1]
```diff
@@ -2,12 +2,12 @@ import express from 'express';
 import bodyParser from 'body-parser';
 import cors from 'cors';
 import path from 'path';
-import fs from 'fs';
 import * as dotenv from 'dotenv';
 import connectDB from './utils/connectDB';
 import userRoute from './routes/user.route';
 import rsvpRoute from './routes/rsvp.route';
 import authRoute from './routes/auth.route';
+import { updateIndexfile } from './utils/IndexFileUtil';
```
Comment (Hyukjoo-Lee) [inline invitation-backend/src/app.ts:9]: It looks great that you have separated the code for updating index into 'IndexFileUtil' file

File: invitation-backend/src/app.ts [hunk 2]
```diff
@@ -20,56 +20,30 @@ app.use(bodyParser.urlencoded({ extended: true }));
...
+});
+
+// This route is only for the og-tag testing using Ngrok.
+app.get('/test-subdomain/:subdomain', (req, res) => {
+  // Development
+  const indexFilePath = path.join(__dirname, '..', '..', 'invitation-frontend', 'public', 'index.html');
+  const { subdomain } = req.params;
+  const updatedIndexFile = updateIndexfile(subdomain, indexFilePath);
+
+  console.log(updatedIndexFile)
```
Comment (Hyukjoo-Lee) [inline invitation-backend/src/app.ts:45]: I want to know where I can see this log

**Review by Hyukjoo-Lee — COMMENTED** [review]
```javascript
// 18 ~ 19 line
app.use(bodyParser.json());
app.use(bodyParser.urlencoded({ extended: true }));
```

I assume that we do not need bodyParser anymore because the recent version of Express include this functionality by default

Please review this reference, and I would be happy to discuss it further our meeting

https://medium.com/@mmajdanski/express-body-parser-and-why-may-not-need-it-335803cd048c

**Review by Hyukjoo-Lee — COMMENTED** [review]
I failed to test og tag using ngrok myself, I hope to try it with you in the today's meeting.

Screen Shot 2023-08-28 at 4 06 22 PM

**Comment by HanKyungSung** [comment]
> ```js
> // 18 ~ 19 line
> app.use(bodyParser.json());
> app.use(bodyParser.urlencoded({ extended: true }));
> ```
> 
> I assume that we do not need bodyParser anymore because the recent version of Express include this functionality by default
> 
> Please review this reference, and I would be happy to discuss it further our meeting
> 
> https://medium.com/@mmajdanski/express-body-parser-and-why-may-not-need-it-335803cd048c

This is a great point. I will take a look and see if we can remove that.

**Comment by HanKyungSung** [comment]
> ```js
> // 18 ~ 19 line
> app.use(bodyParser.json());
> app.use(bodyParser.urlencoded({ extended: true }));
> ```
> 
> I assume that we do not need bodyParser anymore because the recent version of Express include this functionality by default
> 
> Please review this reference, and I would be happy to discuss it further our meeting
> 
> https://medium.com/@mmajdanski/express-body-parser-and-why-may-not-need-it-335803cd048c

Applied new changes that remove the `body-parser`.

**Review by Hyukjoo-Lee — APPROVED** [review]

**Pull request merged**